tiny_dnn/network.h

```diff
@@ -768,7 +778,8 @@ class network {
         for (auto n : net_)
             n->set_parallelize(true);
         optimizer.reset();
-        for (int iter = 0; iter < epoch; iter++) {
+        finish_training_ = false;
+        for (int iter = 0; iter < epoch && !finish_training_; iter++) {
             for (size_t i = 0; i < inputs.size(); i += batch_size) {
```

Comment:
Please also check newly added member variable `finish_training_` at the terminating condition of the inner loop since `finish_ongoing_training` can be called in `on_batch_enumerate`.
```cpp
             for (size_t i = 0; i < inputs.size() && !finish_training_; i += batch_size) {
```